File: java/Program/Laptop.java
// class Laptop merupakan turunan dari class ProdukElektronik
public class Laptop extends ProdukElektronik {
    // atribut class Laptop
    private String tipeProsesor; // intel / amd
    private int kapasitasRAM; // dalam GB
    private int storage; // dalam GB

    // Constructor
    public Laptop() {
    }

    // Getter untuk tipeProsesor
    public String getTipeProsesor() {
        return this.tipeProsesor;
    }

    // Getter untuk kapasitasRAM
    public int getKapasitasRAM() {
        return this.kapasitasRAM;
    }

    // Getter untuk storage
    public int getStorage() {
        return this.storage;
    }

    // Setter untuk tipeProsesor
    public void setTipeProsesor(String tipeProsesor) {
        this.tipeProsesor = tipeProsesor;
    }

    // Setter untuk kapasitasRAM
    public void setKapasitasRAM(int kapasitasRAM) {
        this.kapasitasRAM = kapasitasRAM;
    }

    // Setter untuk storage
    public void setStorage(int storage) {
        this.storage = storage;
    }
}

File: cpp/Program/laptop.cpp
#include <iostream>
#include <string>
#include "produkElektronik.cpp"
using namespace std;

// class laptop turunan dari class ProdukElektronik
class Laptop : public ProdukElektronik {
private:
    // atribut class Laptop
    string tipeProsesor; // seperti intel atau amd
    int kapasitasRAM; // dalam GB
    int storage; // dalam GB

public:
    // Constructor
    Laptop() {
    }

    // Getter untuk tipeProsesor
    string getTipeProsesor() {
        return this->tipeProsesor;
    }

    // Getter untuk kapasitasRAM
    int getKapasitasRAM() {
        return this->kapasitasRAM;
    }

    //  Getter untuk Storage
    int getStorage(){
        return this->storage;
    }

    // Setter untuk tipeProsesor
    void setTipeProsesor(string tipeProsesor) {
        this->tipeProsesor = tipeProsesor;
    }

    // Setter untuk kapasitasRAM
    void setKapasitasRAM(int kapasitasRAM) {
        this->kapasitasRAM = kapasitasRAM;
    }

    // Setter untuk storage
    void setStorage(int storage){
        this->storage = storage;
    }
    
    // Destructor
    ~Laptop() {
    }
};